src/control/lib/spdk/src/nvme_control.c

```diff
@@ -185,20 +195,18 @@ wipe(char *ctrlr_pci_addr)
 			}
 		}
 
-		if (sequence.is_completed != 1) {
+		if (data.was_written != LBA0_WRITE_SUCCESS) {
 			snprintf(ret->info, sizeof(ret->info),
 				 "spdk_nvme_ns_cmd_write() callback\n");
 			ret->rc = -1;
-			spdk_free(sequence.buf);
 			spdk_nvme_ctrlr_free_io_qpair(nentry->qpair);
 			return ret;
 		}
 
 		snprintf(ret->info + sizeof(ret->info),
```

Comment:
I still don't think this is writing to the correct location. Won't this start writing outside the array bounds?

Reply:
I don't get this too.